tests/drizzle/test_drizzle.py

```diff
@@ -1,52 +1,211 @@
+
+    if return_png:
+        # save truth file as figure
+        generate_png(point_function, f"{truth_filename}.png")
+
+    truth_array = np.genfromtxt(f"{truth_filename}.csv", delimiter=",")
+    assert np.allclose(point_function.outsci, truth_array, atol=1e-4)
+
+
+def test_cdriz_edge(
+    point_function, kernel="gaussian", new_truth=True, return_png=True
+):
+    """Similar to test_point_kernel but looking at bright pixels at edge of field. 
+    """
+
+    truth_filename =  f"./tests/drizzle/edge_{kernel}_truth"
+    point_function.insci[0:1, 21:22] = 10
+    cdriz_call(point_function, kernel)
+    if new_truth:
+        save_array(point_function.outsci, f"{truth_filename}.csv")
+    if return_png:
+        generate_png(point_function, f"{truth_filename}.png")
+
+    truth_array = np.genfromtxt(f"{truth_filename}.csv", delimiter=",")
+    assert np.allclose(point_function.outsci, truth_array, atol=1e-4)
+
+
+def test_cdriz_large(
+    point_function, kernel="gaussian", new_truth=True, return_png=True
+):
+    """Similar to test_point_kernel but looking at large pixel. 
+    """
+
+    truth_filename =  f"./tests/drizzle/large_sqaure_{kernel}_truth"
+    point_function.insci[21:25, 22:26] = 10
+    cdriz_call(point_function, kernel)
+    if new_truth:
+        save_array(point_function.outsci, f"{truth_filename}.csv")
+    if return_png:
+        generate_png(point_function, f"{truth_filename}.png")
+
+    truth_array = np.genfromtxt(f"{truth_filename}.csv", delimiter=",")
+    assert np.allclose(point_function.outsci, truth_array, atol=1e-4)
+
+
+def test_cdriz_non_symmetrical(
+    point_function, kernel="gaussian", new_truth=True, return_png=True
+):
+    """Similar to test_point_kernel but looking at large pixel. 
+    """
+
+    truth_filename =  f"./tests/drizzle/nonsymmetrical_{kernel}_truth"
+    point_function.insci[21:25, 22:23] = 10
+    cdriz_call(point_function, kernel)
+    if new_truth:
+        save_array(point_function.outsci, f"{truth_filename}.csv")
+    if return_png:
+        generate_png(point_function, f"{truth_filename}.png")
+
+    truth_array = np.genfromtxt(f"{truth_filename}.csv", delimiter=",")
+    assert np.allclose(point_function.outsci, truth_array, atol=1e-4)
+
+
+@pytest.mark.parametrize("kernel", ["square", "point", "turbo", "gaussian", "lanczos3"])
+def test_zero_input_weight(kernel, point_function):
+    """Tests that do_driz ignores bad pixels, or those that have an input weight (inwht) of 0.
+
+    Parameters
+    ----------
+    kernel : str
+        String associated with one of the c code point kernel options.
+    point_function : Class
+        The Class inintialized in Get_Class which includes all of the inputs need to run cdriz.tdriz.
+    """
+    # add missing/flagged pixels in inwht
+    point_function.inwht[20:21, 22:23] = 0
+
+    # resample
+    cdriz_call(point_function, kernel)
 
     # check that no pixel with 0 weight has any counts:
```

Comment:
If I have understood this correct, I would rephrase to: "Check that any pixel with 0 weight has no counts".

Reply:
good catch, change made.

Reply:
That logic where a pixel with 0 weight has no counts only applies when the `fillval` parameter is set to 0.  I would actually suggest setting `fillval` to "0" instead of "INDEF" for this test for that very reason.

Reply:
I think looking at `outwht` is not that reliable (it _should_ work, in principle). I think, instead, one should test to see that:

1. `outwht` of bad pixels is ether exactly 0 or <1 for border pixels;
2. make a cluster of bad pixels with a high value (1e6). set the values of good pixels to 1 or some random value `0<pixel_val<1`, set `inwht` to 0 for these pixels, and then compute flux of the output image. it should be equal to the flux of all good input pixels.

Reply:
@mcara Thanks for the suggestions, these are really helpful. Ive adapted test_zero_input_weight into the second test that you mention, but I'm still working on understanding the results. 

I still don't understand why the output counts are around half of the input counts after drizzling, but I need to spend some time looking at the algorithm. 

For that first test suggestion, did you mean that bad border pixels with outwht set to zero should have outsci=0? if not, maybe you could elaborate. 

Thanks again!